### PR TITLE
If GEOS is not installed then exclude geos related calls and don't ru…

### PR DESCRIPTION
**Pull request by phoebebright**
…n tests for gis and gis_spatialite

### CODE REVIEW

File: tastypie/resources.py [hunk 1]
```diff
@@ -10,7 +10,14 @@
 import django
 from django.conf import settings
 from django.conf.urls import patterns, url
-from django.contrib.gis.db.models.fields import GeometryField
+
+# Django 1.8 will run without GEOS so fail silently if not installed
+try:
+    from django.contrib.gis.db.models.fields import GeometryField
+    use_geos = True
+except:
```
Comment (SeanHayes) [inline tastypie/resources.py]: Could we except only on the specific error that's raised when GEOS isn't installed?

File: tests/manage_gis.py [hunk 2]
```diff
@@ -2,8 +2,15 @@
...
+    try:
+        from django.contrib.gis.db.models.fields import GeometryField
 
-    execute_from_command_line(sys.argv)
+        os.environ.setdefault("DJANGO_SETTINGS_MODULE", "settings_gis")
+
+        from django.core.management import execute_from_command_line
+
+        execute_from_command_line(sys.argv)
+    except:
```
Comment (SeanHayes) [inline tests/manage_gis.py]: Could we except the specific error here too?

File: tests/manage_gis.py [hunk 3]
```diff
@@ -2,8 +2,15 @@
...
 if __name__ == "__main__":
-    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "settings_gis")
 
-    from django.core.management import execute_from_command_line
+    # Django 1.8 will run without GEOS so don't run tests if not installed
+    try:
+        from django.contrib.gis.db.models.fields import GeometryField
 
-    execute_from_command_line(sys.argv)
+        os.environ.setdefault("DJANGO_SETTINGS_MODULE", "settings_gis")
```
Comment (SeanHayes) [inline tests/manage_gis.py]: Could you move these 3 lines into an "else" block?

File: tastypie/resources.py (hunk 1 above)
Comment (SeanHayes) [inline tastypie/resources.py]: Actually, let's reuse this variable from Django:
https://github.com/django/django/blob/master/django/contrib/gis/geos/__init__.py#L19

**Comment by phoebebright** [comment]
Have updated as you suggest.  Am getting a couple of unrelated errors.
One because (I think) I don't have lxml installed and not sure why the other one.
This version running fine me though.

    (board)iMacs-iMac-2:tests phoebebr$ sh run_all_tests.sh
    ** core **
    Traceback (most recent call last):
      File "./manage_core.py", line 9, in <module>
        execute_from_command_line(sys.argv)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
        utility.execute()
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/__init__.py", line 330, in execute
        self.fetch_command(subcommand).run_from_argv(self.argv)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/commands/test.py", line 30, in run_from_argv
        super(Command, self).run_from_argv(argv)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/base.py", line 393, in run_from_argv
        self.execute(*args, **cmd_options)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/commands/test.py", line 74, in execute
        super(Command, self).execute(*args, **options)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/base.py", line 444, in execute
        output = self.handle(*args, **options)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/commands/test.py", line 90, in handle
        failures = test_runner.run_tests(test_labels)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/test/runner.py", line 209, in run_tests
        suite = self.build_suite(test_labels, extra_tests)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/test/runner.py", line 121, in build_suite
        tests = self.test_loader.loadTestsFromName(label)
      File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/loader.py", line 100, in loadTestsFromName
        parent, obj = obj, getattr(obj, part)
    AttributeError: 'module' object has no attribute 'tests'
    
    
    ** customuser **
    Creating test database for alias 'default'...
    .
    ----------------------------------------------------------------------
    Ran 1 test in 0.042s
    
    OK
    Destroying test database for alias 'default'...
    
    
    ** basic **
    Creating test database for alias 'default'...
    ...................[11/Sep/2015 03:58:50]"GET /api/v1/cached_users/ HTTP/1.1" 200 413
    .[11/Sep/2015 03:58:52]"GET /api/v1/ HTTP/1.1" 400 0
    .[11/Sep/2015 03:58:52]"GET /api/v1/ HTTP/1.1" 200 513
    .[11/Sep/2015 03:58:54]"GET /api/v1/ HTTP/1.1" 500 82070
    F[11/Sep/2015 03:58:55]"GET /api/v1/notes/ HTTP/1.1" 200 676
    .[11/Sep/2015 03:58:56]"GET /api/v1/notes/ HTTP/1.1" 400 34
    .[11/Sep/2015 03:58:57]"POST /api/v1/notes/ HTTP/1.1" 201 0
    [11/Sep/2015 03:58:57]"GET /api/v1/notes/3/ HTTP/1.1" 200 243
    .[11/Sep/2015 03:58:57]"GET /api/v1/private_cached_users/ HTTP/1.1" 200 421
    .[11/Sep/2015 03:58:59]"GET /api/v1/public_cached_users/ HTTP/1.1" 200 420
    .[11/Sep/2015 03:59:00]"GET /api/v1/cached_users/ HTTP/1.1" 200 413
    .
    ======================================================================
    FAIL: test_get_apis_xml (basic.tests.http.HTTPTestCase)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/Users/phoebebr/Development/django-tastypie/tests/basic/tests/http.py", line 65, in test_get_apis_xml
        self.assertEqual(response.status, 200)
    AssertionError: 500 != 200
    
    ----------------------------------------------------------------------
    Ran 29 tests in 10.717s
    
    FAILED (failures=1)
    Destroying test database for alias 'default'...
    
    
    ** alphanumeric **
    Creating test database for alias 'default'...
    ..[11/Sep/2015 03:59:02]"GET /api/v1/ HTTP/1.1" 200 90
    .[11/Sep/2015 03:59:03]"GET /api/v1/ HTTP/1.1" 500 80052
    F[11/Sep/2015 03:59:05]"GET /api/v1/products/ HTTP/1.1" 200 1216
    .[11/Sep/2015 03:59:06]"POST /api/v1/products/ HTTP/1.1" 201 0
    [11/Sep/2015 03:59:06]"GET /api/v1/products/A76124/03/ HTTP/1.1" 200 175
    .
    ======================================================================
    FAIL: test_get_apis_xml (alphanumeric.tests.http.HTTPTestCase)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/Users/phoebebr/Development/django-tastypie/tests/alphanumeric/tests/http.py", line 46, in test_get_apis_xml
        self.assertEqual(response.status, 200, data)
    AssertionError: 
    <!DOCTYPE html>
    <html lang="en">
    <head>
    ...
    
    <div id="summary">
      <h1>ImproperlyConfigured at /api/v1/</h1>
      <pre class="exception_value">Usage of the XML aspects requires lxml and defusedxml.</pre>
      <table class="meta">
    
        <tr>
          <th>Request Method:</th>
          <td>GET</td>
        </tr>
        <tr>
          <th>Request URL:</th>
          <td>http://localhost:8001/api/v1/</td>
        </tr>
    
        <tr>
          <th>Django Version:</th>
          <td>1.8.3</td>
        </tr>
    
        <tr>
          <th>Exception Type:</th>
          <td>ImproperlyConfigured</td>
        </tr>
    
    
        <tr>
          <th>Exception Value:</th>
          <td><pre>Usage of the XML aspects requires lxml and defusedxml.</pre></td>
        </tr>
    
    
        <tr>
          <th>Exception Location:</th>
          <td>/Users/phoebebr/Development/django-tastypie/tests/../tastypie/serializers.py in to_xml, line 377</td>
        </tr>
    
        <tr>
          <th>Python Executable:</th>
          <td>/Users/phoebebr/.virtualenvs/board/bin/python</td>
        </tr>
        <tr>
          <th>Python Version:</th>
          <td>2.7.8</td>
        </tr>
        <tr>
    ...
    
    
    ----------------------------------------------------------------------
    Ran 6 tests in 5.733s
    
    FAILED (failures=1)
    Destroying test database for alias 'default'...
    
    
    ** slashless **
    Creating test database for alias 'default'...
    .
    ----------------------------------------------------------------------
    Ran 1 test in 0.145s
    
    OK
    Destroying test database for alias 'default'...
    
    
    ** namespaced **
    Creating test database for alias 'default'...
    .
    ----------------------------------------------------------------------
    Ran 1 test in 0.226s
    
    OK
    Destroying test database for alias 'default'...
    
    
    ** related **
    Traceback (most recent call last):
      File "./manage_related.py", line 9, in <module>
        execute_from_command_line(sys.argv)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
        utility.execute()
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/__init__.py", line 330, in execute
        self.fetch_command(subcommand).run_from_argv(self.argv)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/commands/test.py", line 30, in run_from_argv
        super(Command, self).run_from_argv(argv)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/base.py", line 393, in run_from_argv
        self.execute(*args, **cmd_options)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/commands/test.py", line 74, in execute
        super(Command, self).execute(*args, **options)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/base.py", line 444, in execute
        output = self.handle(*args, **options)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/core/management/commands/test.py", line 90, in handle
        failures = test_runner.run_tests(test_labels)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/test/runner.py", line 209, in run_tests
        suite = self.build_suite(test_labels, extra_tests)
      File "/Users/phoebebr/.virtualenvs/board/lib/python2.7/site-packages/django/test/runner.py", line 121, in build_suite
        tests = self.test_loader.loadTestsFromName(label)
      File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/loader.py", line 100, in loadTestsFromName
        parent, obj = obj, getattr(obj, part)
    AttributeError: 'module' object has no attribute 'tests'
    
    
    ** validation **
    Creating test database for alias 'default'...
    ........
    ----------------------------------------------------------------------
    Ran 8 tests in 0.317s
    
    OK
    Destroying test database for alias 'default'...
    
    
    ** gis **
    run_all_tests.sh: line 49: createdb: command not found
    GEOS Not installed - not testing gis
    
    
    ** gis_spatialite **
    run_all_tests.sh: line 51: spatialite: command not found
    GEOS Not installed - not testing gis_spatialite
    
    
    ** content_gfk **
    Creating test database for alias 'default'...
    ......
    ----------------------------------------------------------------------
    Ran 6 tests in 0.025s
    
    OK
    Destroying test database for alias 'default'...
    
    
    ** authorization **
    Creating test database for alias 'default'...
    ...........
    ----------------------------------------------------------------------
    Ran 11 tests in 0.595s
    
    OK
    Destroying test database for alias 'default'...

**Comment by SeanHayes** [comment]
I made some changes to this in #1348

**Pull request closed without merging**